services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/resilience/RollbacksSpec.groovy

```diff
@@ -0,0 +1,108 @@
+package org.openkilda.functionaltests.spec.resilience
+
+import static org.openkilda.testing.Constants.PATH_INSTALLATION_TIME
+import static org.openkilda.testing.Constants.WAIT_OFFSET
+
+import org.openkilda.functionaltests.HealthCheckSpecification
+import org.openkilda.functionaltests.extension.failfast.Tidy
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.messaging.info.event.IslChangeType
+import org.openkilda.messaging.info.event.PathNode
+import org.openkilda.messaging.info.event.SwitchChangeType
+import org.openkilda.messaging.payload.flow.FlowState
+import org.openkilda.model.SwitchStatus
+import org.openkilda.testing.model.topology.TopologyDefinition.Isl
+import org.openkilda.testing.model.topology.TopologyDefinition.Switch
+
+import groovy.util.logging.Slf4j
+
+@Slf4j
+class RollbacksSpec extends HealthCheckSpecification {
+
+    @Tidy
+    def "System retries the reroute if it fails to install rules on one the current target path's switches"() {
+        given: "Switch pair with at least 3 available paths, one path should have a transit switch that we will break \
+and at least 1 path must remain safe"
+        List<PathNode> mainPath, failoverPath, safePath
+        Switch switchToBreak //will belong to failoverPath and be absent in safePath
+        Isl islToBreak //will be used to break the mainPath. This ISL is not used in safePath or failoverPath
+        def switchPair = topologyHelper.switchPairs.find { swPair ->
+            if(swPair.paths.size() >= 3) {
+                failoverPath = swPair.paths.find { failoverPathCandidate ->
+                    def failoverSwitches = pathHelper.getInvolvedSwitches(failoverPathCandidate)
+                    safePath = swPair.paths.find { safePathCandidate ->
+                        def safeSwitches = pathHelper.getInvolvedSwitches(safePathCandidate)
+                        switchToBreak = failoverSwitches.find { !(it in safeSwitches) }
+                        def safeIsls = pathHelper.getInvolvedIsls(safePathCandidate).collectMany { [it, it.reversed] }
+                        def failoverIsls = pathHelper.getInvolvedIsls(failoverPathCandidate).collectMany { [it, it.reversed] }
```

Comment:
When looking for islToBreak i want to consider isls in both directions, since I don't want its 'srcSwitch' part to be equal to switchToBreak. For example, if it happens that there is only 1 islToBreak that works for me and its srcSwitch equals to switchToBreak, then I will pick reversed version of this isl